### ISSUE TITLE
unable to choose Proton-Steamplay-version in Wine-Windowslayer for Unix - dropdown empty

### ISSUE THREAD

**Issue by TechGameGeek** [opentrack/opentrack#1896]
Hi,
just compiled opentrack for linux with "wine - windowslayer for unix". It is working great so far e.g. XPlane 12 (linux native), but I am unalbe to pass the trackingdata to steam/proton - games.

As I think to know, I have to choose wine/proton in "wine-windowslayer for unix" and set "Proton Steamplay" to the corresponding version, the game runs with (via dropdown) and give the steam-app-id of the game in the box "steam-app-id"

Unfortunatly my dropdown (Proton Steamplay) is empty.

So I assume I have missed some packages during compiling?

Can you please tell me, what I did wrong to not see "Proton Steamplay" versions?

Thanks!

**Comment by sthalik** [comment]
Probably Valve again broke things.

You can add your path to the list here: https://github.com/opentrack/opentrack/blob/master/proto-wine/proton.cpp#L17-L27.

**Comment by spikerguy** [comment]
> Probably Valve again broke things.

Not really, the path is correct. I can see only few proton versions only.

![image](https://github.com/user-attachments/assets/db0e9190-c26e-4b95-bdb5-9a821d8a620d)


Though I have Proton 8 and 9 also.

**Comment by sthalik** [comment]
Where is the Wine prefix located?

**Comment by spikerguy** [comment]
> > Probably Valve again broke things.
> 
> Not really, the path is correct. I can see only few proton versions only.
> 
> ![image](https://private-user-images.githubusercontent.com/48839029/356817785-db0e9190-c26e-4b95-bdb5-9a821d8a620d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzMDA2OTQsIm5iZiI6MTcyMzMwMDM5NCwicGF0aCI6Ii80ODgzOTAyOS8zNTY4MTc3ODUtZGIwZTkxOTAtYzI2ZS00Yjk1LWJkYjUtOWE4MjFkOGE2MjBkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEwVDE0MzMxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0Yzc2MDgzMGQ2MWQ3ZjQ2NWY5MGYyY2FmOWQ2YTliYmY2MmJmOTM1YTQwZTg3NTE5ZDcwYmJjMzYxMzQzMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wTzWSRFfCKgaOYlb_LWZUF0y78JYepSL1X4qCJ8-w1Y)
> 
> Though I have Proton 8 and 9 also.



> Where is the Wine prefix located?
```
/$USER/Games/SteamLibrary/steamapps/compatdata/805550/pfx/
```

This is game specific, as per the gui wine prefix is supposed to be when wine is used 

for steam proton it just needs proton version and app id, is this information correct ??

I think I will be documenting the whole gaming on linux situation and that way can contribute some documentation for proton use on open track

**Comment by sthalik** [comment]
Try adding it here: https://github.com/opentrack/opentrack/blob/master/proto-wine/proton.cpp#L18-L20

What you need added is probably `/Games/SteamLibrary/steamapps/compatdata/805550/pfx/` as the pathname is preceded by the user's home directory. If that doesn't work then there's some trial and error involved.

This should become fixed once #1879 is implemented (in indeterminate future).

**Comment by spikerguy** [comment]
I see a good documentation [here](https://forum.endeavouros.com/t/linux-gaming-guide/7339)

> Proton engines location:
> 
> ~/.steam/steam/steamapps/common
> 
> Keep in mind that Proton will be installed in the same library as the first Steam Play game you try to launch, and will only be installed in one location.
> 
> Game PREFIX location, where [appid] is your game id (look up [steamdb 25](https://steamdb.info/) if in doubt):
> 
> ~/.steam/steam/steamapps/compatdata/[appid]/pfx/

**Comment by sthalik** [comment]
This is straightforward to implement. 

- find all `steamapps/compatdata` locations, for each of them:
- - find all `appid` subdirectories, for each of them:
- - - find and register all prefix subdirectories

**Comment by spikerguy** [comment]
> This is straightforward to implement.
> 
>     * find all `steamapps/compatdata` locations, for each of them:
> 
>     * * find all `appid` subdirectories, for each of them:
> 
>     * * * find and register all prefix subdirectories

Oh I found Proton 8.0 in different directory

`/home/user/Games/SteamLibrary/steamapps/common/`

so it have proton in compatedata as well as common. this could be the issue. I think a shared memory should resolve it

Thanks

**Comment by spikerguy** [comment]
I applied these patches on opentrack-git from aur and still it is not able to dedect all proton versions

```
--- a/opentrack/proto-wine/proton.cpp	2024-08-10 17:58:40.462649519 +0300
+++ b/opentrack/proto-wine/proton.cpp	2024-08-10 18:03:45.508109393 +0300
@@ -18,6 +18,8 @@
     "/.steam/steam/steamapps/compatdata",
     "/.local/share/Steam/steamapps/compatdata",
     "/.steam/debian-installation/steamapps/compatdata",
+    "/Games/SteamLibrary/steamapps/common",
+    "/Games/SteamLibrary/steamapps/compatdata",
 };
 
 static const char* runtime_paths[] = {
```

also tried 

```
--- a/opentrack/proto-wine/proton.cpp	2024-08-10 17:58:40.462649519 +0300
+++ b/opentrack/proto-wine/proton.cpp	2024-08-10 18:03:45.508109393 +0300
@@ -18,6 +18,8 @@
     "/.steam/steam/steamapps/compatdata",
     "/.local/share/Steam/steamapps/compatdata",
     "/.steam/debian-installation/steamapps/compatdata",
+    "/home/gamer/Games/SteamLibrary/steamapps/common",
+    "home/gamer/Games/SteamLibrary/steamapps/compatdata",
 };
 
 static const char* runtime_paths[] = {
```

does it need proton version and app id both ? or can it just run in appid's pfx ?

**Comment by sthalik** [comment]
As a workaround you can add `/Games/SteamLibrary/steamapps/compatdata/805550`

**Comment by TechGameGeek** [comment]
@sthalik thanks for your reply, I will try it as soon as possible

**Comment by CaptPickguard** [comment]
Running flatpak Steam and having the same issue. Did try compiling manually and adding the paths, but the dropdown never did populate.

Resorted to using the Windows version of Opentrack within the Proton prefix for now.

**Comment by Priton-CE** [comment]
@CaptPickguard for flatpak we had issue #1941 

flatpak is just hard to work with and at least in the past also had other issues.
